addons/sale_coupon/models/sale_order.py

```diff
@@ -84,6 +84,7 @@ def _get_reward_values_product(self, program):
         # Remove needed quantity from reward quantity if same reward and rule product
         if program._is_valid_product(program.reward_product_id):
             reward_product_qty = max_product_qty // (program.rule_min_quantity + program.reward_product_quantity)
+            reward_product_qty *= program.reward_product_quantity
```

Comment:
To make the computation a bit clearer, could you do something like this instead:
```py
# number of times the program should be applied
program_in_order = max_product_qty // (program.rule_min_quantity + program.reward_product_quantity)
# multipled by the reward qty
reward_product_qty = program.reward_product_quantity * program_in_order

Reply:
OK.Thanks!

Reply:
Is sale coupon the enterprise module in V11 and V12?

Reply:
It was in enterprise until saas-12.3 so I guess ok to be fixed here and to be cherry-picked in 11.0 enterprise